rskj-core/src/main/java/co/rsk/cli/tools/StartBootstrap.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2021 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.cli.tools;
+
+import co.rsk.NodeRunner;
+import co.rsk.RskContext;
+import co.rsk.config.InternalService;
+import co.rsk.config.RskSystemProperties;
+import co.rsk.net.discovery.UDPServer;
+import co.rsk.util.NodeStopper;
+import co.rsk.util.PreflightChecksUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Entry point of RSK bootstrap node.
+ *
+ * The bootstrap node starts one service which only participates in the peer discovery protocol. All other capabilities
+ * (which are usually part of a full node, e.g. blocks propagation, syncing, JSON-RPC interface etc.) are disabled.
+ *
+ * Note: this is an experimental tool
+ */
+public class StartBootstrap {
```

Comment:
I would probably start up the `JSON-RPC` service here too, just because I think it is really helpful for debugging and to get information on the nodes. What do you think?

Reply:
> There's a runtime hook in runNode making sure the context is closed on normal shutdown. Actually, I think in case of most exceptions, it will be called twice, but should be ok because it is idempotent.

`preflightChecks.runChecks()` may also crash, which is triggered just before registering a hook. You're right, `close()` can be called multiple times without side effects.

> I would probably start up the JSON-RPC service here too, just because I think it is really helpful for debugging and to get information on the nodes. What do you think?

The ones that provide useful information (e.g. `net_peerCount` and `net_peerList`) rely on `ChannelManager`, which under the hood returns "active" peers (the ones that serve other nodes, e.g. send blocks or txs). The idea here is to avoid usage of such services. I guess, if we really need such debugging capabilities, we may be needing separate methods, which would simply return neighbours of a node (those that are connected via the UDP PING-PONG roundtrip).

Reply:
TIL. I had assumed `peer{Count,List}` returned the peers, active or not. Disregard my comment then 👍🏿